hawkular/client.go

```
@@ -0,0 +1,99 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hawkular
+
+import (
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/hawkular/hawkular-client-go/metrics"
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
+)
+
+type client struct {
+	param         metrics.Parameters
+	hclient       *metrics.Client
+	needMetricDef bool
+}
+
+func newClient(p metrics.Parameters) client {
+	c, err := metrics.NewHawkularClient(p)
+	if err != nil {
+		glog.Fatalln("Initiating hawkular client error: %s", err)
+	}
+	return client{param: p, hclient: c}
+}
+
+func (c client) toHawkular(mts []plugin.Metric) error {
+	mhs := []metrics.MetricHeader{}
+
+	for _, m := range mts {
+		dp, ty := getFloatOnly(m)
+		if ty == metrics.Generic {
+			continue
+		}
+
+		if GClient.needMetricDef {
+			err := c.createMetricDefinition(ty, m.Namespace.Strings(), m.Tags)
+			if err != nil {
+				return err
+			}
+		}
+
+		header := metrics.MetricHeader{
+			ID:   strings.Join(m.Namespace.Strings(), "."),
+			Data: []metrics.Datapoint{dp},
+			Type: ty,
+		}
+		mhs = append(mhs, header)
+		c.hclient.UpdateTags(ty, strings.Join(m.Namespace.Strings(), "."), m.Tags)
+	}
+
+	if len(mhs) > 0 {
+		err := c.hclient.Write(mhs)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c client) createMetricDefinition(t metrics.MetricType, ns []string, tags map[string]string) error {
+	md := metrics.MetricDefinition{
+		Type: t,
+		ID:   strings.Join(ns, "."),
+		Tags: tags,
+	}
+
+	ok, err := c.hclient.Create(md)
+	if !ok && err != nil {
+		return err
+	}
+	return nil
+}
+
+func getFloatOnly(m plugin.Metric) (metrics.Datapoint, metrics.MetricType) {
+	f64, err := metrics.ConvertToFloat64(m.Data)
+	if err != nil {
+		return metrics.Datapoint{}, metrics.Generic
```

This will result in silently dropping all metrics that cannot be converted to a Float64.  Should we at least log (level: warn) something about the error here?